### ISSUE TITLE
No completion data for modules that doesn't respect path

### ISSUE THREAD

**Issue by RUSshy** [d-language-server/dls#63]
I have a reproductive project here: https://github.com/RUSshy/module_dls_bug

I can have completion for ``Data`` located in the ``lib`` subPackage here: https://github.com/RUSshy/module_dls_bug/blob/master/lib/src/repro/lib/data.d wich the module name respect the path of the file

But i can't have completion for ``OtherData`` located in the ``lib`` subPckage here: https://github.com/RUSshy/module_dls_bug/blob/master/lib/src/repro/lib/other.d wich the module name DOESN'T respect the path of the  file

The project compile and run fine

**Comment by RUSshy** [comment]
I run the project like this:

``dub run :app``

**Comment by RUSshy** [comment]
I tried to isolate the issue, and module doesn't need to be in a subPackage, even if a single project, if module name doesn't match filename, it won't be displayed in code completion, even if project compiles

It's been a while i used D, maybe i am doing something wrong

**Comment by LaurentTreguier** [comment]
That's how DCD works; package names have to match the actual path to get completion (if I'm remembering correctly, I haven't done any D in a while either). This is kind of like #58 where module names have to match the file name

**Comment by RUSshy** [comment]
Oh right i forgot about this
Do you know if there is a compiler warning for malformed module names?

**Comment by LaurentTreguier** [comment]
No, I think this is legal in D but DCD simply doesn't support it
